plugins/movetoarchive.koplugin/main.lua

```diff
@@ -0,0 +1,159 @@
+local logger = require("logger")
+local util = require("frontend/util")
+local BaseUtil = require("ffi/util")
+local _ = require("gettext")
+
+local MoveToArchive = WidgetContainer:new{
+    name = "move2archive",
+}
+
+local move_to_archive_settings = LuaSettings:open(("%s/%s"):format(DataStorage:getSettingsDir(), "move_to_archive_settings.lua"))
+local archive_dir_config_key = "archive_dir";
+local last_copied_from_config_key = "last_copied_from_dir";
+local archive_dir_path = move_to_archive_settings:readSetting(archive_dir_config_key)
+local last_copied_from_dir = move_to_archive_settings:readSetting(last_copied_from_config_key)
+
+function MoveToArchive:init()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function MoveToArchive:addToMainMenu(menu_items)
+    menu_items.move_to_archive = {
+        text = _("Move to archive"),
+        sub_item_table = {
+            {
+                text = _("Move current book to archive"),
+                callback = function() self:moveToArchive() end,
+                enabled_func = function() return self:isActionEnabled() end,
+            },
+            {
+                text = _("Copy current book to archive - experimental"),
+                callback = function() self:copyToArchive() end,
+                enabled_func = function() return self:isActionEnabled() end,
+            },
+            {
+                text = _("Go to archive folder"),
+                callback = function()
+                    if not archive_dir_path then
+                        self:showNoArchiveConfirmBox()
+                        return
+                    end
+                    self:openFileBrowser(archive_dir_path)
+                end,
+            },
+            {
+                text = _("Go to last copied/moved from folder"),
+                callback = function()
+                    if not last_copied_from_dir then
+                        UIManager:show(InfoMessage:new{
+                            text = _("No previous folder found.")
+                         })
+                        return
+                    end
+                    self:openFileBrowser(last_copied_from_dir)
+                end,
+            },
+            {
+                text = _("Set archive directory"),
+                keep_menu_open = true,
+                callback =  self.setArchiveDirectory,
+            }
+        },
+    }
+end
+
+function MoveToArchive:moveToArchive()
+    local move_done_text = _("Book moved.\nDo you want to open it from the archive folder?")
+    self:commonProcess(true, move_done_text)
+end
+
+function MoveToArchive:copyToArchive()
+    local copy_done_text =_("Book copied.\nDo you want to open it from the archive folder?")
+    self:commonProcess(false, copy_done_text)
+end
+
+function MoveToArchive:commonProcess(is_move_process, moved_done_text)
+    if not archive_dir_path then
+        self:showNoArchiveConfirmBox()
+        return
+    end
```

Comment:
Dunno if you should not check for `self.archive_dir_path` existence in isActionEnabled().
On the first use, a user may belive the action was done after it has set the archive dir, which it has not. So, may be while the archive dir is not set, have the menu items disabled?